packages/create-astro/src/index.ts

```diff
@@ -78,20 +107,65 @@ export async function main() {
-          delete packageJSON.snowpack; // delete snowpack config only needed in monorepo (can mess up projects)
-          await fs.promises.writeFile(fileLoc, JSON.stringify(packageJSON, undefined, 2));
+  await Promise.all(POSTPROCESS_FILES.map(async (file) => {
+    const fileLoc = path.resolve(path.join(cwd, file));
+
+    switch (file) {
+      case 'CHANGELOG.md': {
+        if (fs.existsSync(fileLoc)) {
+          await fs.promises.rm(fileLoc);
+        }
+        break;
+      }
+      case 'astro.config.mjs': {
+        if (selectedTemplate?.renderers !== true) {
           break;
         }
+        await fs.promises.writeFile(fileLoc, createConfig({ renderers }));
+        break;
       }
-    })
-  );
+      case 'package.json': {
+        const packageJSON = JSON.parse(await fs.promises.readFile(fileLoc, 'utf8'));
+        delete packageJSON.snowpack; // delete snowpack config only needed in monorepo (can mess up projects)
+        // Fetch latest versions of selected renderers
+        const rendererEntries = await Promise.all(['astro', ...renderers].map((renderer: string) => fetch(`https://registry.npmjs.org/${renderer}/latest`).then((res: any) => res.json()).then((res: any) => [renderer, `^${res['version']}`]))) as any;
+        packageJSON.devDependencies = { ...packageJSON.devDependencies ?? {}, ...Object.fromEntries(rendererEntries) }
+        await fs.promises.writeFile(fileLoc, JSON.stringify(packageJSON, undefined, 2));
+        break;
+      }
+    }
+  }));
+
+  // Inject framework components into starter template
+  if (selectedTemplate?.value === 'starter') {
+    let importStatements: string[] = [];
+    let components: string[] = [];
+    await Promise.all(renderers.map(async renderer => {
+      const component = COUNTER_COMPONENTS[renderer as keyof typeof COUNTER_COMPONENTS];
+      const componentName = path.basename(component.filename, path.extname(component.filename));
+      const absFileLoc = path.resolve(cwd, component.filename);
+      importStatements.push(`import ${componentName} from '${component.filename.replace(/^src/, '..')}';`);
+      components.push(`<${componentName}:visible />`);
+      await fs.promises.writeFile(absFileLoc, component.content);
+    }));
+
+    const pageFileLoc = path.resolve(path.join(cwd, 'src', 'pages', 'index.astro'));
+    const content = (await fs.promises.readFile(pageFileLoc)).toString();
+    const lines = content.split('\n');
+    const indent = '        ';
+    const doc = `\n<!-- 
+   - Use imported Framework Components directly in your markup!
+   - 
+   - Note: by default, components are NOT interactive on the client.
+   - The \`:visible\` directive tells Astro to make it interactive.
+   -
+   - See https://github.com/snowpackjs/astro/blob/main/docs/core-concepts/component-hydration.md
+   -->
+`;
+    lines.splice(41, 0, importStatements.length > 0 ? doc.split('\n').map(ln => `${indent}${ln}`).join('\n') : '', ...components.map(ln => `${indent}${ln}`));
```

Comment:
this feels really fragile. Can you do an `indexOf()` search or `replace()` call instead? Something like:

1. Add `<!-- Remember: you can use Frameworks like React & Svelte directly in your markup! -->` to the starter kit `index.astro` file.
2. Replace these hardcoded line numbers with an `indexOf` search or `replace()` call using the entire comment string above.
3. if found, add the correct importStatements

This would let you support other starter kits too, potentially!

Reply:
Very true, I will open a PR with this fix!